### DISCUSSION TITLE
Convert export naming

### DISCUSSION THREAD

**Discussion by hresmaili** [ImageMagick/ImageMagick#5480] [Help] [answered]
Hello,

I am using ImageMagick to convert my .PDF files to .JPG.
I can covert the files without problem, but the name will not be the same as the original files(PDF).
Example: I'll convert PDF01.pdf till PDF15.pdf to JPG and the result is:
PDF0.jpg instead of PDF01.jpg and PDF02.PDF is converted as PDF1.jpg which is not useful.
In fact I want to have exactly the same name for example:  **PDF01.pdf will be convert to PDF01.jpg.**

I am using this syntax:

c:\Users\adm-hresmaili\Desktop\PDF2JPG_test\*pdf c:\Users\adm-hresmaili\Desktop\PDF2JPG_RESULT\PDF%d.jpg

I'll appreciate, if you have some solution like Wild card or other command.
Please see the result in attachment:

![Capture](https://user-images.githubusercontent.com/111859310/186130521-5602870f-d0ba-491d-9745-be99f278f5ec.PNG)

Best regards,

Hamid

**Comment by snibgo** [comment]
I suggest doing it like this:
```
magick *.pdf -set filename:MYNAME "%t" %[filename:MYNAME].jpg
```
If in a Windows BAT script, double the % signs.

On escapes like "%t", see https://imagemagick.org/script/escape.php

**Reply by hresmaili** [reply]
Thank you very much, 

It was very helpful information, however as we use "%t" we can not send the converted files to the destination directory, and the files will remain in the directory of the original files(PDF files). 
I am wondering if I can convert the files from main directory to the specific directory for example using "%d"???

Best,

**Comment by snibgo** [comment, marked as answer]
You can use drive/directory names, exactly as in your OP. So:
```
magick c:\Users\adm-hresmaili\Desktop\PDF2JPG_test\*.pdf -set filename:MYNAME "%t" c:\Users\adm-hresmaili\Desktop\PDF2JPG_RESULT\%[filename:MYNAME].jpg
```

**Reply by hresmaili** [reply]
Thanks again,

It's exactly what I need.

Appreciate,
